### PR TITLE
Prevented duplicate "request completed" logs

### PR DESCRIPTION
**Pull request by chrismeyers**
I noticed that two "request completed" logs were being printed by my Fastify instance following the "incoming request" log.  After setting some breakpoints, I narrowed down the cause of this behavior to this plugin, specifically the `onResponse` hook controlled by the `disposeOnResponse` option.  Removing the `return` from the hook callback stopped the extra log from being printed.  Alternatively, using async/await will also fix this problem.

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: lib/fastifyAwilixPlugin.js [hunk 1]
```diff
@@ -19,13 +19,13 @@ function plugin(fastify, opts, next) {
 
   if (disposeOnClose) {
     fastify.addHook('onClose', (instance, done) => {
-      return diContainer.dispose().then(done)
+      diContainer.dispose().then(done)
```
Comment (Eomm) [inline lib/fastifyAwilixPlugin.js:22]:
```suggestion
      diContainer.dispose().then(done, done)
```

File: lib/fastifyAwilixPlugin.js [hunk 2]
```diff
@@ -19,13 +19,13 @@ function plugin(fastify, opts, next) {
...
     })
   }
 
   if (disposeOnResponse) {
     fastify.addHook('onResponse', (request, reply, done) => {
-      return request.diScope.dispose().then(done)
+      request.diScope.dispose().then(done)
```
Comment (Eomm) [inline lib/fastifyAwilixPlugin.js:28]:
```suggestion
      request.diScope.dispose().then(done, done)
```

**Review by Eomm — COMMENTED** [review]
Would you mind adding a test case?

**Comment by chrismeyers** [comment]
Hi @Eomm, thanks for taking a look at this!  I added a few tests cases to verify the logging is correct.  Let me know if you need anything else.

**Review by zekth — APPROVED** [review]
:v:

**Review by darkgl0w — APPROVED** [review]
LGTM

**Comment by mcollina** [comment]
@kibertoad would you like to land and release this?

**Comment by chrismeyers** [comment]
Hey, just checking in on when this is planned to be merged and released.  Thanks!

**Comment by zekth** [comment]
kind reminder @kibertoad

**Comment by kibertoad** [comment]
Sorry, missed this. On it.

**Review by kibertoad — APPROVED** [review]

**Comment by kibertoad** [comment]
Released in 1.2.1

**Pull request merged**